app/models/course/assessment/question/programming_test_case.rb

```diff
@@ -1,9 +1,7 @@
 # frozen_string_literal: true
 class Course::Assessment::Question::ProgrammingTestCase < ActiveRecord::Base
-  schema_validations except: :description
-
-  validates :description, exclusion: [nil]
-
   belongs_to :question, class_name: Course::Assessment::Question::Programming.name,
                         inverse_of: :test_cases
+
+  enum test_case_type: { private_test: 0, public_test: 1, evaluation_test: 2 }
```

Comment:
Move this before associations, check the attributes order in: https://github.com/Coursemology/coursemology2/blob/master/CONTRIBUTING.md